src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/MasterServiceEventMetricsTests.java

```diff
@@ -0,0 +1,146 @@
+  }
+
+  @Test
+  public void testGetMetricsPath() {
+    String expectedPath = PluginSettings.instance().getMetricsLocation()
+            + PerformanceAnalyzerMetrics.getTimeInterval(startTimeInMills) + "/"
+            + PerformanceAnalyzerMetrics.sThreadsPath + "/" + "thread123" + "/"
+            + PerformanceAnalyzerMetrics.sMasterTaskPath + "/" + "task123" +"/"
+            + PerformanceAnalyzerMetrics.FINISH_FILE_NAME;
+    String actualPath = masterServiceEventMetrics.getMetricsPath(startTimeInMills, "thread123", "task123", PerformanceAnalyzerMetrics.FINISH_FILE_NAME);
+    assertEquals(expectedPath, actualPath);
+
+    try {
+      masterServiceEventMetrics.getMetricsPath(startTimeInMills, "thread123", "task123");
+      fail("Negative scenario test: Should have been a RuntimeException");
+    } catch (RuntimeException ex) {
+      //- expecting exception...2 values passed; 3 expected
+    }
+  }
+
+
+  @Test
+  public void testGenerateFinishMetrics() {
+    assertEquals(-1 ,     masterServiceEventMetrics.lastTaskInsertionOrder);
+    masterServiceEventMetrics.generateFinishMetrics(startTimeInMills);
+
+    masterServiceEventMetrics.lastTaskInsertionOrder = 1;
+    masterServiceEventMetrics.generateFinishMetrics(startTimeInMills);
+    List<Event> metrics = TestUtil.readEvents();
+    String[] jsonStrs = metrics.get(0).value.split("\n");
+    assert jsonStrs.length == 2;
+    assertTrue(jsonStrs[1].contains(AllMetrics.MasterMetricValues.FINISH_TIME.toString()));
+    assertEquals(-1 ,     masterServiceEventMetrics.lastTaskInsertionOrder);
+  }
+
+  @Test
+  public void testCollectMetrics() throws Exception {
+    PrioritizedEsThreadPoolExecutor prioritizedEsThreadPoolExecutor = (PrioritizedEsThreadPoolExecutor) masterServiceEventMetrics
+        .getMasterServiceTPExecutorField().get(ESResources.INSTANCE.getClusterService().getMasterService());
+    SourcePrioritizedRunnable runnable = new SourcePrioritizedRunnable(Priority.HIGH, "_add_listener_") {
+      @Override
+      public void run() {
+        try {
+          Thread.sleep(100L); //dummy runnable
+        } catch (InterruptedException e) {
+        }
+      }
+    };
+
+    prioritizedEsThreadPoolExecutor.submit(runnable);
+    Thread.sleep(1L); // don't delete it
+
+    masterServiceEventMetrics.collectMetrics(startTimeInMills);
+    List<String> jsonStrs = readMetricsInJsonString();
+    assertTrue(jsonStrs.get(0).contains(AllMetrics.MasterMetricDimensions.MASTER_TASK_PRIORITY.toString()));
+    assertTrue(jsonStrs.get(1).contains(AllMetrics.MasterMetricValues.START_TIME.toString()));
+    assertTrue(jsonStrs.get(2).contains(AllMetrics.MasterMetricDimensions.MASTER_TASK_TYPE.toString()));
+    assertTrue(jsonStrs.get(3).contains(AllMetrics.MasterMetricDimensions.MASTER_TASK_METADATA.toString()));
+    assertTrue(jsonStrs.get(4).contains(AllMetrics.MasterMetricDimensions.MASTER_TASK_QUEUE_TIME.toString()));
+  }
+
+
+  private List<String> readMetricsInJsonString() {
```

Comment:
Should we move this method to a util file and reuse it across the tests?

Reply:
Sure, moved to `TestUtil` file